pkg/resources/external_oauth_integration_acceptance_test.go

```diff
@@ -2,186 +2,838 @@ package resources_test
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_snowflake_user_mapping_attribute", string(sdk.ExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeEmailAddress)),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_token_user_mapping_claim.#", "1"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_token_user_mapping_claim.0", "foo"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_jws_keys_url.#", "1"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_jws_keys_url.0", "https://example.com"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_any_role_mode", string(sdk.ExternalOauthSecurityIntegrationAnyRoleModeDisable)),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_allowed_roles_list.#", "1"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_allowed_roles_list.0", role1.Name),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_audience_list.#", "1"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_audience_list.0", "foo"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "external_oauth_scope_delimiter", "."),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "comment", "foo"),
 				),
 			},
+			// change values externally
 			{
-				ResourceName:      "snowflake_external_oauth_integration.test",
-				ImportState:       true,
-				ImportStateVerify: true,
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_ExternalOauthIntegration/completeWithJwsKeysUrlAndAllowedRolesList"),
+				ConfigVariables: m(true),
+				PreConfig: func() {
+					acc.TestClient().SecurityIntegration.UpdateExternalOauth(t, sdk.NewAlterExternalOauthSecurityIntegrationRequest(id).
+						WithSet(*sdk.NewExternalOauthIntegrationSetRequest().
+							WithExternalOauthSnowflakeUserMappingAttribute(sdk.ExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeLoginName),
+						))
+				},
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						planchecks.ExpectDrift("snowflake_external_oauth_integration.test", "external_oauth_snowflake_user_mapping_attribute", sdk.String(string(sdk.ExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeEmailAddress)), sdk.String(string(sdk.ExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeLoginName))),
+
+						planchecks.ExpectChange("snowflake_external_oauth_integration.test", "external_oauth_snowflake_user_mapping_attribute", tfjson.ActionUpdate, sdk.String(string(sdk.ExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeLoginName)), sdk.String(string(sdk.ExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeEmailAddress))),
```

Comment:
could you also assert plancheck.ExpectResourceAction(Update)? Just to make sure

Reply:
Done.